File: code/backend/net.mstoegerer.TaskNest/net.mstoegerer.TaskNest.Api/Domain/DTOs/AttachmentDto.cs
namespace net.mstoegerer.TaskNest.Api.Domain.DTOs;

public class AttachmentDto
{
    public Guid Id { get; set; }
    public string Name { get; set; } = null!;
    public string FileName { get; set; } = null!;
    public string ContentType { get; set; } = null!;
    public byte[] Data { get; set; } = null!;
    public long Size { get; set; }
    public Guid TodoId { get; set; }

    public Guid UploadedById { get; set; }
    public DateTime CreatedUtc { get; set; }
    public DateTime UpdatedUtc { get; set; }
}

File: code/android/app/src/main/java/net/mstoegerer/tasknest/dto/AttachmentDto.kt
package net.mstoegerer.tasknest.dto

import java.util.*

data class AttachmentDto(
    val id: UUID,
    val name: String,
    val fileName: String,
    val contentType: String,
    val data: ByteArray,
    val size: Long,
    val todoId: UUID,
    val uploadedById: UUID,
    val createdUtc: Date,
    val updatedUtc: Date
)